### PR TITLE
Update database.py

### PR DESCRIPTION
**Pull request by aakhiltayal**
Include additional search method for pulling data from materials project

### CODE REVIEW

File: lightshow/database.py [hunk 1]
```diff
@@ -148,11 +161,28 @@ def from_materials_project(cls, **kwargs):
         except KeyError:
             pass
 
-        with MPRester(api_key) as mpr:
```
Comment (matthewcarbone) [inline lightshow/database.py:151]: Why did you get rid of the context manager `with` statement here? This cleans up `mpr` after use and we should use it.

File: lightshow/database.py [hunk 2]
```diff
@@ -148,11 +161,28 @@ def from_materials_project(cls, **kwargs):
...
+        mpr = MPRester(api_key)
+        method = _get_method(kwargs.get("method"), mpr=mpr)
+
+        try:
+            kwargs.pop("method")
+        except KeyError:
+            pass
+
+        if method is not None:
+            searched = getattr(mpr.mateirals, method).search(**kwargs)
```
Comment (matthewcarbone) [inline lightshow/database.py:173]: Typo

**Review by matthewcarbone — CHANGES_REQUESTED** [review]
Tests are failing because of permissions not because I think the code necessarily has a problem.

Can you make sure this passes some of the checks by running the tests locally? There are a ton of them so feel free to just run the tests for a few minutes. There are clearly bugs here that will probably break the tests so that's just a nice sanity check.

Thanks!

File: lightshow/database.py (hunk 1 above)
Comment (aakhiltayal) [reply lightshow/database.py:151]: You are correct now it is changed

**Review by aakhiltayal — COMMENTED** [review]
context manager included

File: lightshow/database.py [hunk 3]
```diff
@@ -161,18 +161,16 @@ def from_materials_project(cls, **kwargs):
...
-
-        if method is not None:
-            searched = getattr(mpr.mateirals, method).search(**kwargs)
-        else:
-            searched = mpr.materials.search(**kwargs)
+        with MPRester(api_key) as mpr:
+            method = _get_method(kwargs.get("method"), mpr=mpr)
+            try:
+                kwargs.pop("method")
+            else:
```
Comment (matthewcarbone) [inline lightshow/database.py:168]: This should be `except KeyError` not `else`.

**Review by matthewcarbone — CHANGES_REQUESTED** [review]
It would also be great to put an example of how this new method can be used in the docstring (which definitely needs some tlc and that's my bad).

File: lightshow/database.py (hunk 3 above)
Comment (aakhiltayal) [reply lightshow/database.py]: solved

**Comment by aakhiltayal** [comment]
> It would also be great to put an example of how this new method can be used in the docstring (which definitely needs some tlc and that's my bad).

**Comment by aakhiltayal** [comment]
> It would also be great to put an example of how this new method can be used in the docstring (which definitely needs some tlc and that's my bad).

I agree please feel free to modify the code as you fit good.

**Comment by matthewcarbone** [comment]
@aakhiltayal this definitely needs an example somewhere. From your updated docstring I'm still not exactly sure how you plan on using this! Can you post a code snippet on how you do it here? I can then merge this and make some quick corrections myself after.

**Pull request merged**